### PR TITLE
added option to set status without beeing admin

### PR DESCRIPTION
**Pull request by nicoknoll**

### CODE REVIEW

File: features/users/server/methods.coffee [hunk 1]
```diff
@@ -210,7 +210,7 @@ Meteor.methods {
 
     'users.setStatus': (id, status) ->
 
-        if not Roles.userIsInRole Meteor.userId(), 'admin'
+        if not Roles.userIsInRole(Meteor.userId(), 'admin')
```
Comment (enricofoschi) [inline features/users/server/methods.coffee]: no need to add parenthesis here

File: features/users/server/methods.coffee [hunk 2]
```diff
@@ -222,6 +222,17 @@ Meteor.methods {
                 status: status
         }
 
+    'users.setStatusNoSecurityCheck': (id, status) ->
```
Comment (enricofoschi) [inline features/users/server/methods.coffee]: Wow! Why do we want such a thing?

Comment (enricofoschi) [inline features/users/server/methods.coffee]: This opens quite a big security flaw: any user can update any user status, regardless of their permissions. Why do we want such a thing?

**Comment by enricofoschi** [comment]
change it to user.setCurrentUserStatus and use Meteor.userId()

File: features/users/server/methods.coffee [hunk 3]
```diff
@@ -222,6 +222,20 @@ Meteor.methods {
                 status: status
         }
 
+    'users.setCurrentUserStatus': (id, status) ->
+
+        if not Roles.userIsInRole(Meteor.userId(), 'admin') and Meteor.userId() isnt id
```
Comment (enricofoschi) [inline features/users/server/methods.coffee]: Don't pass the id - just use it in the method :)

**Comment by nicoknoll** [comment]
Going to handle it out of crater.js.

**Pull request closed without merging**